torchaudio/_backend/utils.py

```diff
@@ -0,0 +1,292 @@
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import BinaryIO, Dict, Optional, Tuple, Union
+
+import torch
+from torchaudio.backend.common import AudioMetaData
+from torchaudio.io._compat import (
+    info_audio,
+    info_audio_fileobj,
+    load_audio,
+    load_audio_fileobj,
+    save_audio,
+    save_audio_fileobj,
+)
+from torchaudio._extension import _FFMPEG_INITIALIZED, _SOX_INITIALIZED
+from torchaudio.utils.sox_utils import get_buffer_size
+import torchaudio.backend.soundfile_backend as soundfile_backend
+
+
+class Backend(ABC):
+    @staticmethod
+    @abstractmethod
+    def info(uri: Union[BinaryIO, str], format: Optional[str]) -> AudioMetaData:
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def load(
+        uri: Union[BinaryIO, str],
+        frame_offset: int = 0,
+        num_frames: int = -1,
+        normalize: bool = True,
+        channels_first: bool = True,
+        format: Optional[str] = None,
+    ) -> Tuple[torch.Tensor, int]:
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def save(
+        uri: Union[BinaryIO, str],
+        src: torch.Tensor,
+        sample_rate: int,
+        channels_first: bool = True,
+        format: Optional[str] = None,
+    ) -> None:
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def can_decode(uri, format) -> bool:
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def can_encode(uri, format) -> bool:
+        raise NotImplementedError
+
+
+class FFmpegBackend(Backend):
+    @staticmethod
+    def info(uri: Union[BinaryIO, str], format: Optional[str]) -> AudioMetaData:
+        if hasattr(uri, "read"):
+            buffer_size = get_buffer_size()
```

Comment:
How about making buffer_size a keyword argument?
I think it's confusing that this value is configurable via sox_utils though it's ffmpeg.